syntax = "proto3";

message AccountCreationRequest {
     int32 buyer_id = 1;
     string buyer_name = 2;
     string buyer_email = 3;
     string buyer_username = 4;
     string buyer_password = 5;
     int32 items_purchased = 6;
     string buyer_country=7;
     string buyer_city=8;
     string buyer_contact=9;
     string buyer_address=10;
     string buyer_zipcode=11;
     string is_active=12;
}

message AccountLoginRequest {
     string buyer_username = 3;
     string buyer_password = 4;
}

message AccountCreationResponse {
     string status = 1;
}

message Accounts {
     repeated AccountCreationRequest accounts = 1;
}

message GetAccount {
     string acc = 1;
}

message SearchProductRequest {
     string product = 1;
}

message SearchProductResponse {
     repeated ProductDetails products = 1;
}

message Keyword {
     string words = 1;
}

message GetProduct {}

message ProductDetails {
     int32 productId = 1;
     string product_name = 2;
     string price = 3;
     int32 quantity = 4;
     int32 seller_id = 5;
     int32 category = 6;
     string category_name = 7;
     repeated Keyword keywords = 8;
}

message AddToCartRequest {
     int32 productInCart = 2;
     string price = 3;
     int32 quantityInCart = 4;
     int32 userInCart = 5;
}

message AddToCartResponse {
     string status = 1;
     string price = 2;
}

service BuyerActions {
    rpc createAccount (AccountCreationRequest) returns (AccountCreationResponse);
    rpc login (AccountLoginRequest) returns (AccountCreationRequest);
    rpc search (SearchProductRequest) returns (SearchProductResponse);
    rpc getProducts (GetProduct) returns (SearchProductResponse);
    rpc addToCart (GetProduct) returns (SearchProductResponse);
}